validation/url_checker.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validates schema urls."""
+import csv
+import git
+import os
+import pathlib
+import re
+import requests
+
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+flags.DEFINE_boolean("update_allowlist", False,
+                     "If set, regenerates url_allowlist.csv.")
+
+API_ERROR = "An HTTP {} code was returned by the autopush API. Please ensure that DC_API_KEY is set to the autopush API key."
+API_PREFIX = "https://autopush.api.datacommons.org/v2/node?key="
```

Comment:
Is the use of autopush to ensure recent schema?

Reply:
Yep, assuming that we'd want to catch these early. One tradeoff is that a user would need to know the autopush API key, so if that becomes a problem (depending on who would run this test or while automating it) I can revert to the prod api

Reply:
Good for now, and we can make the env a flag.